autoload/clap/provider/bcommits.vim

```diff
@@ -5,13 +5,78 @@ let s:save_cpo = &cpoptions
 set cpoptions&vim
 
 let s:bcommits = {}
+let s:bcommits.syntax = 'clap_diff'
+
+let s:current = bufname(g:clap.start.bufnr)
+let s:begin = '^[^0-9]*[0-9]\{4}-[0-9]\{2}-[0-9]\{2}\s\+'
 
 function! s:bcommits.source() abort
-  call g:clap.abort('Not implemented yet')
+  let s:git_root = clap#path#get_git_root()
+  if empty(s:git_root)
+    call g:clap.abort('Not in git repository')
+    return
+  endif
+
+  let s:current = bufname(g:clap.start.bufnr)
+  let managed = 0
+  if !empty(s:current)
+    call system('git show '.s:current.' 2> '.(has('win32') ? 'nul' : '/dev/null'))
```

Comment:
I doubt `2> nul` can work on Windows.

Reply:
Should be able to redirect. But I can not verify :-(

Reply:
I don't have a Windows machine either. I'm unwilling to merge a patch that possibly can't work on Windows. Need someone to verify that.

Reply:
Tried on a win10 VM, did not install vim, only tried git.  2> nul works fine

```
c:\>git show dumpfile
fatal: not a git repository (or any of the parent directories): .git

c:\>git show dumpfile 2> nul

c:\>
```